### PR TITLE
feature: add insurance flow

### PR DESCRIPTION
**Pull request by LBokan**
### Description

Please provide a brief description of the changes you have made in your PR.

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [x] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots of the changed components here.

#### Screenshots/Video after change
insurance-landing-page
insurance-assignment-page

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/8e1e178d-5ff3-4c52-905e-5cdf6e708194

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/b791a45f-88a2-45ab-a8fb-9ee44defe09e

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/9ea8a677-53de-4c2a-81d5-5dc0313b42eb

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/09d143fe-c9dd-4e25-be8c-a3b459c0f93b

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/fd665139-2384-4ef6-bc35-4e74b578b0bd

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/3ab5efd4-7ca4-41de-8ea1-51b03073302c

### CODE REVIEW

File: frontend/src/components/AssignmentDialog/AssignmentDialog.vue [hunk 1]
```diff
@@ -104,6 +144,9 @@
 
   const { assignment } = storeToRefs(useAssignmentStore());
   const { closeAndResetAssignmentModalOrDialog } = useAssignmentStore();
+  const { contact } = storeToRefs(useContactStore());
+  const { phoneNumber } = storeToRefs(usePhoneNumberStore());
+  const { address } = storeToRefs(useAddressStore());
```
Comment (rmaskalenka) [inline frontend/src/components/AssignmentDialog/AssignmentDialog.vue:149]: so you are storing all store values as refs just to get 1 flag from each of them? This shounds like an overkill

File: frontend/src/components/AssignmentDialog/AssignmentInfoCard.vue [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+  const openSetRepairFacilityModal = () => {
+    isSetRepairFacilityModalOpen.value = true;
+  };
+
+  const closeSetRepairFacilityModal = () => {
+    isSetRepairFacilityModalOpen.value = false;
+  };
+
+  const handleBusComments = (busEvent: string) => {
```
Comment (rmaskalenka) [inline frontend/src/components/AssignmentDialog/AssignmentInfoCard.vue:206]: why string and not boolean? String checks are much slower than boolean

File: frontend/src/components/AssignmentDialog/ContactsInfoCard.vue [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+              <v-icon color="white" icon="mdi-pencil" />
+              <v-tooltip
+                activator="parent"
+                text="Edit contact"
+                location="bottom"
+              />
+            </v-btn>
+
+            <v-btn
+              v-if="userState.role == ROLES.insurance && !isExcludedContactTypes(contact.type)"
```
Comment (rmaskalenka) [inline frontend/src/components/AssignmentDialog/ContactsInfoCard.vue:28]:
```suggestion
              v-if="userState.role === ROLES.insurance && !isExcludedContactTypes(contact.type)"

```

Please use tripple equality instead

File: frontend/src/components/CommentModal.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  const emits = defineEmits(['close-modal']);
+  const props = defineProps({
+    commentData: {
+      type: Object,
+      default: () => {}
+    }
+  });
+  const commentValue = ref(props.commentData.text);
+  
+  const { userState } = storeToRefs(useUserStore());
```
Comment (rmaskalenka) [inline frontend/src/components/CommentModal.vue:57]: if you only interested in role you can have a `computed` variable with role instead

**Review by rmaskalenka — CHANGES_REQUESTED** [review]

File: frontend/src/components/AssignmentDialog/AssignmentDialog.vue (hunk 1 above)
Comment (LBokan) [reply frontend/src/components/AssignmentDialog/AssignmentDialog.vue:149]: Done: removed it

File: frontend/src/components/AssignmentDialog/AssignmentInfoCard.vue (hunk 2 above)
Comment (LBokan) [reply frontend/src/components/AssignmentDialog/AssignmentInfoCard.vue:206]: Done

File: frontend/src/components/AssignmentDialog/ContactsInfoCard.vue (hunk 3 above)
Comment (LBokan) [reply frontend/src/components/AssignmentDialog/ContactsInfoCard.vue:28]: Done

File: frontend/src/components/CommentModal.vue (hunk 4 above)
Comment (LBokan) [reply frontend/src/components/CommentModal.vue:57]: Done

**Review by rmaskalenka — APPROVED** [review]

**Pull request merged**